analyses/tp53_nf1_score/08_survival_TP53.Rmd

```diff
@@ -0,0 +1,231 @@
+---
+title: "tp53_survival_analysis"
+author: "Zhuangzhuang Geng"
+date: "2023-09-20"
+output: html_document
+---
+
+
+## load libraries
+
+```{r libraries}
+suppressWarnings({
+  library(tidyverse)
+  library(survival)
+  library(ggplot2)
+  library(patchwork)
+})
+```
+
+## set directories adn source functions for plotting 
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analyses_dir <- file.path(root_dir, "analyses", "tp53_nf1_score")
+results_dir <- file.path(analyses_dir, "results")
+plots_dir <- file.path(analyses_dir, "plots")
+
+## source this script for plotting
+source(file.path(analyses_dir, "util", "survival_models.R"))
+```
+
+```{r}
+HGG_survival <- read_tsv(file.path(results_dir, "HGG_survial_tp53.tsv"))
+
+HGG_survival <- HGG_survival %>%
+  mutate(OS_years = (OS_days / 365.25), 
+         binned_tp53 = replace_na(binned_tp53, "no tp53 score"), 
+         plot_mol_subtype = case_when(grepl("IHG", molecular_subtype) ~ "IHG", 
+                                      TRUE ~ molecular_subtype))
+
+```
+
+## plot the Kaplan–Meier survival curve on 
+## broad_mol, mol_subtype_without_tp53, gender, binned age and binned tp53 score
+
+```{r}
+var <- c("reported_gender", "binned_age", "binned_tp53", "broad_mol", "mol_subtype_without_tp53", "race")
+
+for(i in var){
+  
+  kap_OS <- survival_analysis(
+  metadata = HGG_survival, 
+  ind_var = i,
+  test = "kap.meier",
+  metadata_sample_col = "sample_id",
+  days_col = "OS_days",
+  status_col = "OS_status"
+  )
+print(i)
+  plot <- plotKM(model = kap_OS, 
+       variable = i, 
+       combined = FALSE, 
+       title = i)
+
+  if(i == "race"){
+    ggsave(file.path(plots_dir, paste0("KM_", i, ".pdf")), plot, 
+       width = 16, height = 6, units = "in", 
+       device = "pdf")
+  }else{
+    ggsave(file.path(plots_dir, paste0("KM_", i, ".pdf")), plot, 
+       width = 10, height = 6, units = "in", 
+       device = "pdf")
+  }
+  
+}
+
+```
+
+##
+
+```{r}
+
+HGG_survival <- HGG_survival %>% 
```

Comment:
for this model, the ages are out of numerical order and 0,15 is not the control as with the other plots - can you make the plots consistent with age control?